src/librustc_typeck/diagnostics.rs

```diff
@@ -170,6 +171,37 @@ Since `return;` is just like `return ();`, there is a mismatch between the
 function's return type and the value being returned.
 "##,
 
+E0070: r##"
+The left-hand side of an assignment operator must be an lvalue expression. An
+lvalue expression represents a memory location and includes item paths (ie,
+namespaced variables), dereferences, indexing expressions, and field
+references.
```

Comment:
The `includes` in this sentence tripped me up slightly. Perhaps something like the following?

> An lvalue expression represents a memory location and can be a variable, a dereference, an indexing expression or a field reference.

You could write `variable (with optional namespacing)` to be more precise, but I think it's ok relying on the reference.